post-processing/readme.qmd

```diff
@@ -0,0 +1,169 @@
+
+### Current list of post-processors
+
+* Calibration of probability estimates to increase their probabilistic fidelity to the observed data.
+
+```r
+# 'object' is a pre-made calibration object
+# Data Inputs: class probabilities
+# Data Outputs: class probabilities and recomputed class predictions
+add_prob_calibration(object, .priority = 1.0)
+```
+
+* For binary classification outcomes, there may be the need to optimize the probability threshold that produces hard class predictions.
+
+```r
+# Potentially tunable
+# Data Inputs: class probabilities
+# Data Outputs: class predictions
+add_prob_threshold(threshold = numeric(), .priority = 2.0)
+```
+
+* An addition of an equivocal zone where highly uncertain probability estimates where no hard class prediction is reported. 
+
+```r
+# Potentially tunable
+# Data Inputs: class probabilities
+# Data Outputs: class predictions
+add_cls_eq_zone(value = numeric(), threshold = numeric(), .priority = 3.0)
+```
+
+* Calibration of regression predictions to avoid regression to the mean or other systematic issues with models.
+
+```r
+# 'object' is a pre-made calibration object
+# Data Inputs: regression predictions
+# Data Outputs: regression predictions
+add_reg_calibration(object, .priority = 1.0)
+```
+
+* A general "mutate" operation where users can do on-the-fly operations using self-contained functions. For example,
+
+  - If a user log-transformed their data prior to modeling, they can exponentiate here. 
+  - For regression models, the predictions can be thresholded to be within a specific range. 
+
+```r
+# User will have to always set the priority
+# Data Inputs: all predictions
+# Data Outputs: all predictions (only these columns are retained)
+add_post_mutate(..., .priority)
+```
+
+This is likely to be a fairly complete menu of possible options. For example, at a later date, nearest-neighbor adjustment of regression predictions (see [this blog post](https://rviews.rstudio.com/2020/05/21/modern-rule-based-models/)) would be an interesting option.
+
+## Implementation
+
+There are implementation issues within the workflows and tune packages to discuss. There is also the idea that a small side-package will keep the underlying functions and data structures (for now at least). We'll tentatively call this the `dwai` package (pronounced "duh-why" and short for Don't Worry About It).
+
+### dwai
+
+This package will contain
+
+ - A container for the list of possible post-processors specified by the user.
+ - A validation system to resolve conflicts in type or priority. 
+ - An interface to apply the operations to the predicted values. 
+ - The requisite package dependencies (primarily the probably package).
+ 
+The dwai code may eventually make its way into workflows or probably. 
```

Comment:
If this is our plan, I might argue we put this functionality in workflows or probably from the get-go. Feels a bit like this living its own package could eventually feel like technical debt.